components/modelregistry/modelregistry.go

```diff
@@ -82,6 +85,46 @@ func (m *ModelRegistry) ReconcileComponent(_ context.Context, cli client.Client,
 				return fmt.Errorf("failed to update image from %s : %w", Path, err)
 			}
 		}
+
+		// Create model registry viewer and editor ClusterRoles if not present
+		// These roles are meant for user convenience to limit access to model-registry CRs and don't affect model-registry-operator
+		// Hence they are created here, and won't be deleted when the model registry component is Removed
+		_, err = cluster.CreateClusterRole(cli, "modelregistry-viewer-role",
+			[]v1.PolicyRule{
+				{APIGroups: []string{"modelregistry.opendatahub.io"}, Resources: []string{"modelregistries"},
+					Verbs: []string{"get", "list", "watch"}},
+				{APIGroups: []string{"modelregistry.opendatahub.io"}, Resources: []string{"modelregistries/status"},
+					Verbs: []string{"get"}},
+			},
+			cluster.WithLabels(labels.ODH.Component(ComponentName), "true",
+				labels.ODH.ComponentName, ComponentName,
+				"app.kubernetes.io/name", "clusterrole",
+				"app.kubernetes.io/instance", "modelregistry-viewer-role",
+				"app.kubernetes.io/component", "rbac",
+				"app.kubernetes.io/created-by", "model-registry-operator",
+				"app.kubernetes.io/part-of", "model-registry-operator",
+				"app.kubernetes.io/managed-by", "model-registry-operator"))
+		if err != nil {
+			return err
+		}
+		_, err = cluster.CreateClusterRole(cli, "modelregistry-editor-role",
+			[]v1.PolicyRule{
+				{APIGroups: []string{"modelregistry.opendatahub.io"}, Resources: []string{"modelregistries"},
+					Verbs: []string{"create", "delete", "get", "list", "patch", "update", "watch"}},
+				{APIGroups: []string{"modelregistry.opendatahub.io"}, Resources: []string{"modelregistries/status"},
+					Verbs: []string{"get"}},
+			},
+			cluster.WithLabels(labels.ODH.Component(ComponentName), "true",
+				labels.ODH.ComponentName, ComponentName,
+				"app.kubernetes.io/name", "clusterrole",
+				"app.kubernetes.io/instance", "modelregistry-editor-role",
+				"app.kubernetes.io/component", "rbac",
+				"app.kubernetes.io/created-by", "model-registry-operator",
+				"app.kubernetes.io/part-of", "model-registry-operator",
+				"app.kubernetes.io/managed-by", "model-registry-operator"))
```

Comment:
Maybe we can add those labels to `pkg/metadata/labels` and re-use? I think this will be used more moving forward, so extracting it to constant might help.

https://github.com/opendatahub-io/opendatahub-operator/blob/08b74a7ce1dd0e005151893159d0f3a8846c3c87/pkg/metadata/labels/types.go#L13-L17